consumer/src/main/kotlin/streams/procedures/StreamsSinkProcedures.kt

```diff
@@ -0,0 +1,73 @@
+package streams.procedures
+
+import org.apache.commons.lang3.StringUtils
+import org.neo4j.logging.Log
+import org.neo4j.procedure.*
+import streams.StreamsEventConsumerFactory
+import streams.StreamsEventSinkConfigMapper
+import streams.StreamsSinkConfiguration
+import java.util.stream.Stream
+
+class StreamResult(@JvmField val event: Any)
+
+class StreamsSinkProcedures {
+
+    @JvmField @Context
+    var log: Log? = null
+
+    @Procedure(mode = Mode.SCHEMA, name = "streams.consume")
+    @Description("streams.consume(topic, config) - Allows to subscribe custom topics")
+    fun consume(@Name("topic") topic: String?,
+                @Name(value = "config", defaultValue = "{}") config: Map<String, Any>?): Stream<StreamResult> {
+        checkEnabled()
+        if (topic == null || topic == StringUtils.EMPTY) {
+            log?.info("Topic empty, no message sent")
+            return Stream.empty()
+        }
+
+        val properties = if (config == null) {
```

Comment:
you can use `config?.mapValues { it.value.toString() } ?: emptyMap()`

Reply:
Ok!